src/runtime-rs/src/shim/src/task_service.rs

```diff
@@ -0,0 +1,848 @@
+                    Ok(status) => status.exit_code as u32,
+                };
+
+                stdin = match &container.common_process.stdin {
+                    Some(stdin) => stdin.clone(),
+                    None => String::from(""),
+                };
+
+                stdout = match &container.common_process.stdout {
+                    Some(stdout) => stdout.clone(),
+                    None => String::from(""),
+                };
+
+                stderr = match &container.common_process.stderr {
+                    Some(stderr) => stderr.clone(),
+                    None => String::from(""),
+                };
+
+                rsp.set_terminal(container.common_process.terminal);
+            } else if let Some(exec) = container.processes.get(eid) {
+                match exec.state {
+                    State::Ready => rsp.set_status(Status::CREATED),
+                    State::Running => rsp.set_status(Status::RUNNING),
+                    State::Stopped => rsp.set_status(Status::STOPPED),
+                    State::Paused => rsp.set_status(Status::PAUSED),
+                }
+
+                exit_code = match exec.common_process.status.try_read() {
+                    Err(_e) => 0,
+                    Ok(status) => status.exit_code as u32,
+                };
+
+                stdin = match &exec.common_process.stdin {
+                    Some(stdin) => stdin.clone(),
+                    None => String::from(""),
+                };
+
+                stdout = match &exec.common_process.stdout {
+                    Some(stdout) => stdout.clone(),
+                    None => String::from(""),
+                };
+
+                stderr = match &exec.common_process.stderr {
+                    Some(stderr) => stderr.clone(),
+                    None => String::from(""),
+                };
+            }
+
+            rsp.set_stdin(stdin);
+            rsp.set_stdout(stdout);
+            rsp.set_stderr(stderr);
+        } else {
+            warn!("failed to fine container {} process {}", cid, eid);
+            rsp.set_status(Status::STOPPED);
+        }
+
+        rsp.set_exit_status(exit_code);
+
+        Ok(rsp)
+    }
+
+    fn do_create(&self, req: shim::CreateTaskRequest) -> Result<shim::CreateTaskResponse> {
+        let rsp = shim::CreateTaskResponse::new();
+        let s = self.sandbox.clone();
+        let mut sandbox = s.lock().unwrap();
+
+        let (mounts, stdin, stdout, stderr) = {
+            let mut mounts = vec![];
+            for mount in req.get_rootfs() {
+                mounts.push(virtcontainers::container::Mount {
+                    fs_type: String::from(mount.get_field_type()),
+                    source: String::from(mount.get_source()),
+                    destination: String::from(mount.get_target()),
+                    options: mount.get_options().to_vec(),
+                    device_id: String::new(),
+                    // TODO:
```

Comment:
This looks like it could get "lost" unless we add a GitHub issue URL here (and a unit test to check it).

Reply:
Removed unused TODO and corresponding code in the first revision.  Add them back in future release if necessary